### PR TITLE
Remove Sentry.HTTPClient.child_spec/0

### LINKED ISSUE #587: Make `Sentry.HTTPClient.child_spec/1` callback be optional
My use-case is I want to use Tesla as an HTTP Client and use pre-configure Finch that is already defined in the application tree. But the `HTTPClient` requires the implementor to implement `child_spec`. 

My proposal is to make `child_spec` optional and attach it to the supervisor if it's available. I can help working on this if you're ok with this proposal. :)

Thanks

### PR DESCRIPTION
**Pull request by wingyplus**
Fixes #587

### CODE REVIEW

**Comment by whatyouhide** [comment]
I'm wondering if we should instead just remove the `child_spec/0` callback altogether, and have users start their HTTP client in their own supervision tree.

Thoughts?

**Comment by wingyplus** [comment]
> 

@whatyouhide I'm ok with this but it's ok to break the hackney configuration?

**Comment by whatyouhide** [comment]
@wingyplus we don't need to break the hackney configuration. We can do something like

```elixir
client = Config.client()

children = [Task...]

children =
  if client == Sentry.HackneyClient do
    children ++ [Sentry.HackneyClient]
  else
    children
  end
```

**Comment by wingyplus** [comment]
@whatyouhide Ahh, I see. PR updated. 🙇

File: lib/sentry.ex [hunk 1]
```diff
@@ -304,11 +304,19 @@ defmodule Sentry do
...
 
-    if Config.client() == Sentry.HackneyClient do
+    client = Config.client()
+
+    children =
+      if client == Sentry.HackneyClient do
+        children ++ [Sentry.HackneyClient.child_spec()]
+      else
+        children
+      end
```
Comment (wingyplus) [inline lib/sentry.ex:317]: I think it should be move under hackney check.

**Comment by wingyplus** [comment]
@whatyouhide what else I need do to make it merge?

**Comment by whatyouhide** [comment]
Hey @wingyplus I’m figuring out whether the best strategy is to make this an optional callback or go with [my suggestion here](https://github.com/getsentry/sentry-elixir/pull/589#issuecomment-1672920725). I'll need some time since there are other things to figure out first in this lib 😄

**Comment by wingyplus** [comment]
> Hey @wingyplus I’m figuring out whether the best strategy is to make this an optional callback or go with [my suggestion here](https://github.com/getsentry/sentry-elixir/pull/589#issuecomment-1672920725). I'll need some time since there are other things to figure out first in this lib 😄

Understood. I'm ok with both directions. Let me know which one you prefer. 🙇

**Comment by whatyouhide** [comment]
@wingyplus I was heads down on Sentry stuff and ended up doing this myself, but it's your original solution with the optional callback. Thanks! 💟

**Pull request closed without merging**